Game/ConsoleApplication2/MovingGameObject.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "GameObject.h"
+#include "Constants.h"
+
+
+class MovingGameObject : public GameObject
+{
```

Comment:
This class doesn't cause any issues, but beware of overusing inheritance just because its there. This behavior is too simple in my opinion for the performance impact of polymorphism. 

Creating a pure virtual "think" function in an AIObject class which can be overridden by derived classes makes more sense to me. If you don't want to rewrite the code in every object I think a static AIManager which has a "LinearSeek(Vec& pos, const Vec& target, float dt)" and "ApplyAcceleration(Vec& velocity, Vec& accel, float dt)" makes more sense. That way all AI is in the AIManager, considering AI is so important in this project I suggest separating it out completely into helper functions. Will come in handy for ARGO as well I imagine